src/fsharp/PrettyNaming.fs

```diff
@@ -318,38 +458,35 @@ module internal Microsoft.FSharp.Compiler.PrettyNaming
 
     let private chopStringTo (s:string) (c:char) =
         (* chopStringTo "abcdef" 'c' --> "def" *)
-        if s.IndexOf c <> -1 then
-            let i =  s.IndexOf c + 1
+        match s.IndexOf c with
+        | -1 -> s
+        | idx ->
+            let i = idx + 1
             s.Substring(i, s.Length - i)
-        else
-            s
 
     /// Try to chop "get_" or "set_" from a string
     let TryChopPropertyName (s: string) =
-        // extract the logical name from any mangled name produced by MakeMemberDataAndMangledNameForMemberVal 
-        let s = 
-            if s.StartsWith("get_", System.StringComparison.Ordinal) || 
-               s.StartsWith("set_", System.StringComparison.Ordinal) 
-            then s 
-            else chopStringTo s '.'
-
-        if s.Length <= 4 || (let s = s.Substring(0,4) in s <> "get_" && s <> "set_") then
-            None
+        // extract the logical name from any mangled name produced by MakeMemberDataAndMangledNameForMemberVal
+        if s.Length <= 4 then None
         else 
-            Some(s.Substring(4,s.Length - 4) )
-
-
+            let s =
+                if s.StartsWith("get_", System.StringComparison.Ordinal) ||
+                   s.StartsWith("set_", System.StringComparison.Ordinal)
+                then s
+                else chopStringTo s '.'
+
+            if s.StartsWith("get_", System.StringComparison.Ordinal) ||
+               s.StartsWith("set_", System.StringComparison.Ordinal)
+            then Some (s.Substring(4, s.Length - 4))
+            else None
```

Comment:
Can we simplify this further, so there is only a single check for "get_" and "set_" prefixes?

```fsharp
    /// Try to chop "get_" or "set_" from a string
    let TryChopPropertyName (s: string) =
        // extract the logical name from any mangled name produced by MakeMemberDataAndMangledNameForMemberVal
        if s.Length <= 4 then None else
        let s = chopStringTo s '.'
        if s.StartsWith("get_", System.StringComparison.Ordinal) ||
           s.StartsWith("set_", System.StringComparison.Ordinal)
            then Some (s.Substring(4, s.Length - 4))
            else None
```